common/src/main/scala/org/apache/celeborn/common/authentication/Credential.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.celeborn.common.authentication
+
+trait PasswordCredential {
+  def username: String
+  def password: String
+  def extraInfo: Map[String, String] = Map.empty
+}
+
+case class DefaultPasswordCredential(
+    username: String,
+    password: String,
+    override val extraInfo: Map[String, String] = Map.empty) extends PasswordCredential
+
+trait TokenCredential {
+  def token: String
+  def extraInfo: Map[String, String] = Map.empty
+}
+
+case class DefaultTokenCredential(
+    token: String,
+    override val extraInfo: Map[String, String] = Map.empty) extends TokenCredential
+
+object Credential {
+  val CLIENT_IP_KEY = "clientIp"
```

Comment:
mistake, should I rename it to `CLIENT_IP`? @pan3793

Reply:
Okay. BTW, what's the open api generated style for parameter name? "clientIp" or "client_ip"? let's sync them too.

Reply:
```
@javax.annotation.Generated(value = "org.openapitools.codegen.languages.JavaClientCodegen", date = "2024-07-10T15:33:39.089-07:00[America/Los_Angeles]")
public class WorkersResponse {
  public static final String JSON_PROPERTY_WORKERS = "workers";
  private List<WorkerData> workers = new ArrayList<>();

  public static final String JSON_PROPERTY_LOST_WORKERS = "lostWorkers";
  private List<WorkerTimestampData> lostWorkers;
```

We can keep it as it is now.